scripts/nn/examples/mnist_lenet_distrib_sgd.dml

```diff
@@ -134,12 +134,15 @@ train = function(matrix[double] X, matrix[double] Y,
       db4_agg = matrix(0, rows=parallel_batches, cols=nrow(b4)*ncol(b4))
 
       # Run graph on each mini-batch in this group in parallel (ideally on multiple GPUs)
-      parfor (j in 1:parallel_batches) {
+      # mode=REMOTE_SPARK, opt=CONSTRAINED
+      parfor (j in 1:parallel_batches, mode=REMOTE_SPARK, opt=CONSTRAINED, resultmerge=REMOTE_SPARK) {
         # Get a mini-batch in this group
-        beg = ((j-1) * batch_size) %% nrow(X_group_batch) + 1
-        end = min(nrow(X_group_batch), beg + batch_size - 1)
-        X_batch = X_group_batch[beg:end,]
-        y_batch = y_group_batch[beg:end,]
+        batch_beg = group_beg + ((j-1) * batch_size) %% row_group + 1
```

Comment:
Also, I don't think the `+ 1` is needed here since `group_beg` already contains it and we're collapsing the indexing expressions into one.